src/odemis/acq/milling/test/patterns_test.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+"""
+@author: Patrick Cleeve
+
+Copyright © 2024, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import logging
+import unittest
+import json
+import numpy
+from odemis.acq.milling.patterns import RectanglePatternParameters, TrenchPatternParameters, MicroexpansionPatternParameters
+
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)-15s: %(message)s")
+logging.getLogger().setLevel(logging.DEBUG)
+
+class MillingPatternParamersTestCase(unittest.TestCase):
```

Comment:
I would create a separate class per Parameter (rectangle, trench, microexpansion), and then a separate test case for each block that you now added (test_assignment, test_to_json, ...).

Reply:
ok done